File: Java version/app/src/main/java/it/unical/mat/embasp/specializations/dlvhex/DLVHEXAnswerSets.java
package it.unical.mat.embasp.specializations.dlvhex;

import it.unical.mat.embasp.languages.asp.AnswerSets;
import it.unical.mat.parsers.asp.ASPSolversParser;

public class DLVHEXAnswerSets extends AnswerSets {
	public DLVHEXAnswerSets(final String answerSets) {
		super(answerSets);
	}

	public DLVHEXAnswerSets(final String output, final String errors) {
		super(output, errors);
	}

	@Override
	protected void parse() {
		ASPSolversParser.parseDLVHEX(this, output, true);
	}
}

File: Python version/specializations/dlvhex/dlvhex_answer_sets.py
from languages.asp.answer_sets import AnswerSets
from parsers.asp.asp_solvers_parser import ASPSolversParser


class DLVHEXAnswerSets(AnswerSets):
    def __init__(self, output, errors=None):
        super(DLVHEXAnswerSets, self).__init__(output, errors)

    def _parse(self):
        ASPSolversParser.parse_dlvhex(self, self._output, True)
